### PR TITLE
Handle points

### PR DESCRIPTION
**Pull request by DucAnhPhi**
- classify commata
- classify multiply sign
- fix minor bugs and improve reliability

### CODE REVIEW

File: contour.py [hunk 1]
```diff
@@ -71,10 +71,17 @@ def is_inside_y_range(y):
...
+        if self.trueWidth > self.trueHeight * 2:
+            if len(self.holes) == 0:
+                area = cv.contourArea(self.contour)
+                minRectArea = self.trueHeight * self.trueWidth
+                extent = float(area)/minRectArea
+                if extent > 0.7 or self.trueWidth > self.trueHeight * 4:
+                    return True
+
+    def is_vertical_bar(self):
+        if self.width < self.height:
```
Comment (The-Silverowl) [inline contour.py]: Vielleicht wäre es gut den Winkel zu bestimmen.

File: segmentation.py [hunk 2]
```diff
@@ -93,14 +93,51 @@ def handle_nested_contour(self, cnt, index):
...
+        yDevToCentroid = abs(currCnt.center[1] - preCnt.center[1])
+        yDevToBottom = abs(currCnt.center[1] - preCnt.y2)
+        if yDevToBottom < yDevToCentroid:
+            if cv.contourArea(currCnt.contour) < cv.contourArea(preCnt.contour):
+                currCnt.set_bar_type(BarType.COMMA)
+
+    def label_multiply(self, currCnt, preCnt):
+        yDevToCentroid = abs(currCnt.center[1] - preCnt.center[1])
+        xDevToCentroid = abs(currCnt.center[0] - preCnt.center[0])
+        if 2 * yDevToCentroid < xDevToCentroid:
```
Comment (The-Silverowl) [inline segmentation.py]: Why not add yDevToBottom > yDevToCentroid ?

File: segmentation.py [hunk 3]
```diff
@@ -93,14 +93,51 @@ def handle_nested_contour(self, cnt, index):
...
+
+    def label_comma_and_multiply(self, lines):
+        def label_signs(orderedContours):
+            for i in range(len(orderedContours)):
+                if i == 0:
+                    continue
+                currCnt = orderedContours[i]
+                preCnt = orderedContours[i-1]
+                self.label_comma(currCnt, preCnt)
+                self.label_multiply(currCnt, preCnt)
```
Comment (The-Silverowl) [inline segmentation.py]: Both of these could be made more consistent, if checked over 3 contours (e.g. 1*1 or 1,1).

File: app.py [hunk 4]
```diff
@@ -57,14 +56,11 @@ def process(self, frame, name="TrainingSamples/Image_"):
         preprocessed = cv.cvtColor(preprocessed, cv.COLOR_GRAY2BGR)
 
         lines = LineOrdering(filtered).get_lines(frame)
-        for line in lines:
```
Comment (The-Silverowl) [inline app.py]: I am just testing your code, and I am in awe.
You did this in a single day? And it works so well.
(Sure there are some issues, but still...)
Consider me impressed.

File: segmentation.py (hunk 3 above)
Comment (DucAnhPhi) [reply segmentation.py]: true but how :D

File: segmentation.py (hunk 2 above)
Comment (DucAnhPhi) [reply segmentation.py]: also valid. I am not sure why I did it like this might work as well

**Pull request merged**